src/python/pants/backend/python/goals/pytest_runner.py

```diff
@@ -204,6 +214,31 @@ def _count_pytest_tests(contents: DigestContents) -> int:
     return sum(len(_TEST_PATTERN.findall(file.content)) for file in contents)
 
 
+async def validate_pytest_cov_included(_pytest: PyTest):
+    if _pytest.requirements:
+        # We'll only be using this subset of the lockfile.
+        requirements = {PipRequirement.parse(req) for req in _pytest.requirements}
+    else:
+        # We'll be using the entire lockfile.
+        lockfile_metadata = await get_lockfile_metadata(_pytest)
+        if not isinstance(lockfile_metadata, (PythonLockfileMetadataV2, PythonLockfileMetadataV3)):
```

Comment:
This isn't future proof to V4+. Maybe invert the conditional to check if its V1?

Reply:
Good idea, done.

Reply:
Oh, hmm, mypy doesn't like it (which I guess was why I did it this way to begin with). We don't know if a hypothetical V4 will have a `requirements` attribute (I mean, we informally know it will, but the type system doesn't). I could squelch the check, but I think it is actually saying something informative in this case. So I think I'd prefer to leave this as-is. In any case, we should deprecate V1 (and maybe V2) I think...